### PR TITLE
feat(insured-bridge): add caching of addresses and common params to save gas in relaying

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

The insured bridge will often call out to other contracts during the live cycle of a relay to fetch information. This includes getting the latest `OptimisticOracle` and `Store` addresses from the `Finder` as well as fetching the `proposerBondPct`, `identifier` and `optimisticOracleLiveness` values from the `BridgeAdmin`. This is somewhat inefficient and it would save gas if the `BridgePool` Stored these locally.

**Summary**

This PR refactors the bridge pool to store the aforementioned variables locally so the contract does not have to call out to external contracts during the bridging action. To facilitate updates of these variables, two new methods are added: 
1. `syncWithBridgeAdminParams` to pull latest params from the `BridgePool` and
2. `syncWithFinderAddresses` to pull latest addresses from the `Finder`

Collectively, these two changes save the following gas vs the code before this PR:
1. `relayDeposit`: before PR: 542890. After PR: 525371. Net saving: 17519
2. `speedUpRelay`: before PR: 74884. After PR: 74929. Net saving: -45
3. `settleRelay`: before PR: 308940. After PR: 297025. Net saving: 11915

Total gas saving for a full relay life cycle: 29389

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 1]
```diff
@@ -58,6 +58,15 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
     // Administrative contract that deployed this contract and also houses all state variables needed to relay deposits.
     BridgeAdminInterface public bridgeAdmin;
 
+    // Store local instances of the contract instances to save gas relaying. Can be sync with the Finder at any time.
+    StoreInterface public store;
+    OptimisticOracleInterface public optimisticOracle;
+
+    // Store local instances of contract params to save gas relaying. Can be sync with the BridgeAdmin at any time.
```
Comment (nicholaspai) [inline packages/core/contracts/insured-bridge/BridgePool.sol:65]:
```suggestion
    // Store local instances of contract params to save gas relaying. Can be synced with the BridgeAdmin at any time.
```

File: packages/core/contracts/insured-bridge/BridgePool.sol [hunk 2]
```diff
@@ -617,44 +644,35 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
...
     function _proposeOraclePriceRelay(
         uint256 amount,
         uint256 requestTimestamp,
         bytes memory customAncillaryData
     ) private {
-        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
-        uint256 proposerBondPct =
-            FixedPoint.Unsigned(uint256(bridgeAdmin.proposerBondPct())).div(FixedPoint.fromUnscaledUint(1)).rawValue;
-        uint256 finalFee = _getStore().computeFinalFee(address(l1Token)).rawValue;
+        uint256 proposerBondPct_ =
```
Comment (nicholaspai) [inline packages/core/contracts/insured-bridge/BridgePool.sol:669]: I like the underscore _ before the variable name, wdyt?

**Review by nicholaspai — APPROVED** [review]
This is awesome, should bring the average and best cast cases down

File: packages/core/contracts/insured-bridge/BridgePool.sol (hunk 2 above)
Comment (chrismaree) [reply packages/core/contracts/insured-bridge/BridgePool.sol:669]: sure, updated.

**Pull request merged**